buildSrc/src/main/kotlin/java-library-conventions.gradle.kts

```diff
@@ -173,7 +173,7 @@ val compileModule by tasks.registering(JavaCompile::class) {
 tasks.compileJava {
 	// See: https://docs.oracle.com/en/java/javase/11/tools/javac.html
 	options.compilerArgs.addAll(listOf(
-			"-Xlint", // Enables all recommended warnings.
+			"-Xlint:all,-deprecation", // Enables all recommended warnings.
```

Comment:
Do we need to do this for all projects?

Reply:
No. It was a convenient place to redeem all temporarely introduced deprecation warnings at once.

Would we ship 5.5-M2 with the deprecated feature and remove it before 5.5 GA?

Reply:
No, the whole point of deprecating it first was to give users a chance to stop using it before we actually delete it. Thus, I think we should not remove it before 5.6/

Reply:
Alright. Then let's keep `-deprecation` active until `5.5 GA` is out and remove it immediately after starting with `5.6-SNAPSHOT`.

Reply:
Fair enough, please create an issue for us to do this for 5.6 M1.

Reply:
Done: https://github.com/junit-team/junit5/issues/1902